test/integration/config.test.js

```diff
@@ -0,0 +1,224 @@
+        expect(mockServer.isDone()).to.be.true;
+        helper.assertions.assertUserProjectSetup(defaultExpectedUser, defaultExpectedProject, cb);
+      });
+    });
+
+    // Scenario: CLI prompts for email and password. Verified but 2FA is on. Prompts for 2FA token - a wrong one is entered.
+    // Prompts again for token - provide a correct token. Provide correct project info.
+    // Expects user and project to be set properly.
+    it('with valid input and 2FA required should set user and project', (cb) => {
+      mockServer.loginWithTwoFactorAuthFail();
+      mockServer.loginForSuccess(fixtureUser.existentWith2FA);
+
+      mockServer.apps();
+      mockServer.dataLinks();
+
+      const twoFactorTokenStub = sandbox.stub(prompt, 'getTwoFactorToken');
+      twoFactorTokenStub.callsArgWith(1, null, fixtureUser.invalidTwoFactorToken);
+      twoFactorTokenStub.callsArgWith(1, null, fixtureUser.validTwoFactorToken);
+
+      require(cmdConfigPath)(null, command, (err) => {
+        expect(err).to.not.exist;
+        assertPromptStubsForSuccess();
+        expect(mockServer.isDone()).to.be.true;
+        helper.assertions.assertUserProjectSetup(defaultExpectedUser, defaultExpectedProject, cb);
+      });
+    });
+
+    // User/session info must be set even though no internal dataLinks (Flex services) are found.
+    it('with valid input and no Flex services created should set user and return error', (cb) => {
+      mockServer.loginForSuccess();
+      mockServer.apps();
+      mockServer.dataLinks([]);
+
+      require(cmdConfigPath)(null, command, (err) => {
+        helper.assertions.assertError(err, Errors.NoFlexServicesFound);
+        assertPromptStubsForSuccess(false);
+
+        expect(mockServer.isDone()).to.be.true;
+
+        const expectedProject = null;
+        helper.assertions.assertUserProjectSetup(defaultExpectedUser, expectedProject, cb);
+      });
+    });
+  });
+
+  describe('without explicit args and credentials from environment', () => {
+    before(() => {
+      helper.setup.projectPromptStubsForSuccess(sandbox);
+    });
+
+    afterEach(() => {
+      helper.env.unsetCredentials();
+    });
+
+    after(() => {
+      sandbox.restore();
+    });
+
+    it('with valid project input and valid credentials should set user and project', (cb) => {
+      mockServer.loginForSuccess();
+      mockServer.apps();
+      mockServer.dataLinks();
+
+      helper.env.setCredentials(fixtureUser.existent.email, fixtureUser.existent.password);
+
+      require(cmdConfigPath)(null, command, (err) => {
+        expect(err).to.not.exist;
+        expect(mockServer.isDone()).to.be.true;
+        helper.assertions.assertUserProjectSetup(defaultExpectedUser, defaultExpectedProject, cb);
+      });
+    });
+
+    // TODO: The config command should fail right after credentials turn out to be invalid. But a few more requests are
+    // made and that's why the test fails. Probably, a fix should be applied to the code.
```

Comment:
Should you create a ticket if the work won't be done in this branch?

Reply:
Well, if you agree that it's sensible to be fixed? It could be done during the refactoring.

Reply:
Up to you. I'm suggesting to make a ticket now to track the work if you decide that this action-item should be addressed at a later time.